File: Java/Sqrt(x).java
// 1ms, 100.00%; 36.4MB, 5.00%
class Solution {
    public static int mySqrt(int x) {
        if (x < 2) {
            return x;
        }
        int cur = (x >> 1), next;
        while (cur > (next = (cur + x / cur) >> 1)) {
            cur = next;
        }
        return cur;
    }
}

File: Go/Sqrt(x).go
// 0ms, 100.00%; 2.2MB, 83.33%
func mySqrt(x int) int {
	if x < 2 {
		return x
	}
	res := x >> 1
	for {
		tmp := (res + x/res) >> 1
		if tmp >= res {
			break
		}
		res = tmp
	}
	return res
}
